src/mqt/yaqs/simulator.py

```
@@ -217,6 +227,215 @@ def _digital_worker(
+        if key not in sampled_by_id:
+            sampled_by_id[key] = noise_model.sample(rng=rng)
+        return sampled_by_id[key]
+
+    sampled_default = sample(compiled.default_noise_model)
+    instructions = tuple(
+        replace(instruction, noise_model=sample(instruction.noise_model)) for instruction in compiled.instructions
+    )
+    return replace(compiled, instructions=instructions, default_noise_model=sampled_default)
+
+
+def _execute_program_trajectory(
```

Ruff (0.16.1)

[warning] Too many statements (64 > 50)

(PLR0915)